src/CameraComponent_V4L2.cpp

```diff
@@ -0,0 +1,467 @@
+    v4l2_close_device(fd);
+
+    strcpy((char *)camInfo.vendorName, "Intel");
+    strcpy((char *)camInfo.modelName, "RealSense R200");
+    camInfo.firmware_version = 1;
+    camInfo.focal_length = 0;
+    camInfo.sensor_size_h = 0;
+    camInfo.sensor_size_v = 0;
+    camInfo.resolution_h = 0;
+    camInfo.resolution_v = 0;
+    camInfo.lens_id = 0;
+    camInfo.flags = ~0u;
+    camInfo.cam_definition_version = 1;
+}
+
+void CameraComponent_V4L2::initStorageInfo()
+{
+    // TODO:: Fill storage details with values
+}
+
+void CameraComponent_V4L2::initSupportedValues()
+{
+    // TODO :: Get the details of the supported list from the camera device or from xml/conf file
+}
+
+void CameraComponent_V4L2::initDefaultValues()
+{
+    // TODO :: Get the details of the default values from the camera device and then save it in DB
+    saveParameter(CameraParameters::CAMERA_MODE,
+                  (uint32_t)CameraParameters::id_camera_mode::ID_CAMERA_MODE_VIDEO);
+    saveParameter(CameraParameters::BRIGHTNESS, (uint32_t)56);
+    saveParameter(CameraParameters::CONTRAST, (uint32_t)32);
+    saveParameter(CameraParameters::SATURATION, (uint32_t)128);
+    saveParameter(CameraParameters::HUE, (int32_t)0);
+    saveParameter(CameraParameters::WHITE_BALANCE_MODE,
+                  (uint32_t)CameraParameters::id_white_balance_mode::ID_WHITE_BALANCE_AUTO);
+    saveParameter(CameraParameters::GAMMA, (uint32_t)220);
+    saveParameter(CameraParameters::GAIN, (uint32_t)32);
+    saveParameter(CameraParameters::POWER_LINE_FREQ_MODE, (uint32_t)0);
+    saveParameter(CameraParameters::WHITE_BALANCE_TEMPERATURE, (uint32_t)6500);
+    saveParameter(CameraParameters::SHARPNESS, (uint32_t)0);
+    saveParameter(CameraParameters::BACKLIGHT_COMPENSATION, (uint32_t)1);
+    saveParameter(CameraParameters::EXPOSURE_MODE, (uint32_t)3);
+    saveParameter(CameraParameters::EXPOSURE_ABSOLUTE, (uint32_t)1);
+    saveParameter(CameraParameters::VIDEO_SIZE,
+                  (uint32_t)CameraParameters::id_video_size::ID_VIDEO_SIZE_640x480x30);
+
+    // dummy values
+    saveParameter(CameraParameters::IMAGE_SIZE,
+                  CameraParameters::id_image_size::ID_IMAGE_SIZE_3264x2448);
+    saveParameter(CameraParameters::IMAGE_FORMAT,
+                  CameraParameters::id_image_format::ID_IMAGE_FORMAT_JPEG);
+    saveParameter(CameraParameters::PIXEL_FORMAT,
+                  CameraParameters::id_pixel_format::ID_PIXEL_FORMAT_YUV420P);
+}
+
+int CameraComponent_V4L2::getParam(const char *param_id, char *param_value)
+{
+    // TODO :: Do appropriate checks
+    // query the value set in the map and fill the output, return appropriate value
+    std::string value = camParam.getParameter(param_id);
+    strcpy(param_value, value.c_str());
+    return 1;
+}
+
+int CameraComponent_V4L2::setParam(const char *param_id, const char *param_value, int param_type)
+{
+    int ret = 1;
+    mavlink_param_union_t u;
+    memcpy(&u.param_float, param_value, sizeof(float));
```

Comment:
This looks like a possible overflow or invalid bytes copied when size of the param_value is < than sizeof(float). Is this going to work for uint8?